Intersect.Client/Entities/Player.cs

```diff
@@ -1534,57 +1549,104 @@ public bool TryBlock()
 
         public bool TryAttack()
         {
-            if (AttackTimer > Timing.Global.Milliseconds || IsBlocking || (IsMoving && !Options.Instance.PlayerOpts.AllowCombatMovement))
+            if (AttackTimer > Timing.Global.Milliseconds || IsBlocking ||
+                (IsMoving && !Options.Instance.PlayerOpts.AllowCombatMovement))
             {
                 return false;
             }
 
             int x = Globals.Me.X;
             int y = Globals.Me.Y;
             var map = Globals.Me.MapId;
+
+            CheckAdjacentTiles();
+
             switch (Globals.Me.Dir)
             {
-                case 0:
+                case (byte)Directions.Up:
                     y--;
-
                     break;
-                case 1:
-                    y++;
 
+                case (byte)Directions.Down:
+                    y++;
                     break;
-                case 2:
+
+                case (byte)Directions.Left:
                     x--;
+                    break;
 
+                case (byte)Directions.Right:
+                    x++;
                     break;
-                case 3:
+
+                case (byte)Directions.UpLeft:
+                    y--;
+                    x--;
+                    break;
+
+                case (byte)Directions.UpRight:
+                    y--;
                     x++;
+                    break;
+
+                case (byte)Directions.DownLeft:
+                    y++;
+                    x--;
+                    break;
 
+                case (byte)Directions.DownRight:
+                    y++;
+                    x++;
                     break;
             }
 
             if (TryGetRealLocation(ref x, ref y, ref map))
             {
+                // Iterate through all entities
                 foreach (var en in Globals.Entities)
                 {
-                    if (en.Value == null)
+                    // Skip if the entity is not within the player's map, is null or is the current player.
+                    if (en.Value == null || en.Value == Globals.Me || en.Value.MapId != map)
                     {
                         continue;
                     }
 
-                    if (en.Value != Globals.Me)
+                    if (TargetBox.MyEntity != null)
```

Comment:
So several things:

1. CanBeAttacked was completely removed from here, and it should not have been.
2. Please change CanBeAttacked from a method with overrides to a getter-only virtual property with overrides.
3. if !CanBeAttacked then continue comes _before_ this line.

Reply:
![imagen](https://user-images.githubusercontent.com/17498701/215359955-bdc03eca-edc1-4d57-996a-98717c4f896e.png)


![imagen](https://user-images.githubusercontent.com/17498701/215359947-9224dc94-1b54-4826-b849-892801c43155.png)